### PR TITLE
Fixed Algorithm Optimization and Night Mode Redundancy

### PR DESCRIPTION
**Pull request by Aarthif-Nawaz**
## Purpose / Description
Fixed Algorithm Optimization using bitwise "&" operator instead of "=="
Fixed Checking of NightMode redundancy

## Fixes
Fixes _Link to the issues._

## Approach
Faster Performance In both new code change

## How Has This Been Tested?

Tested in my redmi note 9 pro. A bit more faster than before

## Learning (optional, can help others)
_Describe the research stage_

_Links to blog posts, patterns, libraries or addons used to solve this problem_

## Checklist
_Please, go through these checks before submitting the PR._

- [V ] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [ V] You have a descriptive commit message with a short title (first line, max 50 chars).
- [ V] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [ V] You have commented your code, particularly in hard-to-understand areas
- [ V] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/CardUtils.java [hunk 1]
```diff
@@ -54,7 +54,7 @@ public static void markAll(List<Note> notes, boolean mark) {
 
     public static boolean isIn(long[] array, long val) {
         for (long v : array) {
-            if (v == val) {
+            if (v & val) {
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/CardUtils.java:57]: This doesn't compile

File: AnkiDroid/src/main/java/com/ichi2/anki/cardviewer/CardAppearance.java [hunk 2]
```diff
@@ -48,7 +48,7 @@ public boolean isNightMode() {
      */
     public boolean hasUserDefinedNightMode(Card card) {
         // TODO: find more robust solution that won't match unrelated classes like "night_mode_old"
-        return card.css().contains(".night_mode") || card.css().contains(".nightMode");
+        return card.css().contains(".night");
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/cardviewer/CardAppearance.java:51]: This match is too permissive

**Review by david-allison-1 — COMMENTED** [review]
Hi, thanks for the contribution!

It'd be best if you started off with one of our "Good First Issues". Sadly, neither of these fixes are appropriate for the codebase

https://github.com/ankidroid/Anki-Android/issues?q=is%3Aissue+is%3Aopen+label%3A%22Good+First+Issue%21%22+no%3Aassignee

**Comment by mikehardy** [comment]
Additionally - performance enhancements in general won't be accepted without before + after timings showing the impact over a few test runs under real usage with a collection you actually study

This is a high standard to meet on purpose - we want to spend effort on things users care about. "performance gains" you can't measure while doing tasks users do are not something users care about at all.

**Pull request closed without merging**